src/Microsoft.AspNet.Mvc.Core/ModelBinding/MutableObjectModelBinder.cs

```diff
@@ -40,18 +40,22 @@ public class MutableObjectModelBinder : IModelBinder
                 return null;
             }
 
-            EnsureModel(bindingContext);
-            var result = await CreateAndPopulateDto(bindingContext, mutableObjectBinderContext.PropertyMetadata);
+            // Create model first (if necessary) to avoid reporting errors about properties when activation fails.
+            var model = GetModel(bindingContext);
+
+            var results = await BindPropertiesAsync(bindingContext, mutableObjectBinderContext.PropertyMetadata);
```

Comment:
I'm curious: What's the perf advantage? Is `CreatePropertyModelName()` expensive for example? Could see a larger advantage if we stopped binding after the first error.